add resize_widget_dimensions() to change widget dimensions after gridster initialization

hrosenbauer opened the pull request:
(this is a updated version of my first pull-request - now in a separate branch)
(this pull request is based on the discussion of https://github.com/ducksboard/gridster.js/pull/77 and the code of https://gist.github.com/OwlyCode/6421823 with small changes, so most of the credits go to https://github.com/OwlyCode)

This will add resize_widget_dimensions() to gridster.js which allows to resize the widget dimensions after initialization:

``` javascript
// initialize gridster with 200x200px widgets using a 5px margin
var gridster = $('.gridster ul').gridster({
    widget_base_dimensions: [200, 200],
    widget_margins: [5, 5]
}).data('gridster');

// resize the widgets to 300x300px and 10px margin
gridster.resize_widget_dimensions({
    widget_base_dimensions: [300, 300],
    widget_margins: [10, 10]
});
```

It's very useful to resize the dashboard when the browser window is resized.

hrosenbauer committed:
add resize_widget_dimension()
resize widgets after initialization

hrosenbauer committed:
add resize_widget_dimension()
resize widgets after initialization

Conflicts:
	src/jquery.gridster.js

skozz commented:
+1

kwak27 committed:
fix(gridster): fix add_widget to use correct size_y when adding rows

dlabaj commented:
Exactly what we needed.

dlabaj commented:
Found one potential issue.  After the grid has been render to the DOM and is showing on the web page, I am listening for when the browser window is resized.  If I set the max_cols value (in my case 2), and don't set autogrow_cols to true when all the widgets are moved to the first column, you can't move them back to the second column.  Other than that everything else has been working great.

hrosenbauer committed:
fix(draggable): fix offset during drag

when the scrollX/Y is changed during drag (e.g. by the mousewheel)
the offset wasn't correct

hrosenbauer committed:
fix(draggable): replaced scrollX/Y with scrollLeft/Top

window.scrollX/Y is only available in Chrome, not in IE

hrosenbauer committed:
fix min-width and min-height after resize

remove min-width and min-height after stopping resize

hrosenbauer committed:
fix min-width and min-height after resize

remove min-width and min-height after stopping resize

dasmall committed:
fix(gridster): Removing previously added style tags before adding new one.

Hopefully fixes #211 and #294.

hrosenbauer committed:
fix(draggable): bind drag events to $document

$body won't receive mouseup if the cursor is outside the browser window, $document does

Misiu commented:
:+1:

rgomesf commented:
Nice feature!

vieron committed:
fix(draggable): handle both touch and click events

Fixes #207, #236, #329, and #380

vieron committed:
Release 0.5.2

vieron committed:
chore: move sorting methods in prototype to the constructor

Useful to use them when we there is no gridster instance, like sorting
a previously stored output from the serialize method.

vieron committed:
style: some formatting changes

vieron committed:
feat(gridster): move widget up when added if there is space available

vieron committed:
fix(gridster): sort widgets appropriately when reading them from DOM

vieron committed:
feat: make gridster AMD compatible

vieron committed:
feature(gridster): added `auto_init` config option, default to true

Useful when using `positionschanged` event, fired before gridster
instance is cached.

vieron committed:
chore: add reference to defaults in constructors

vieron committed:
fix(gridster): custom `ignore_dragging` overwrites the default value

hrosenbauer committed:
Merge branch 'resize-widget-dimensions' of https://github.com/rebugger/gridster.js into resize-widget-dimensions

Conflicts:
	src/jquery.gridster.js

ChristopheBelpaire commented:
+1 !!

OleMchls commented:
:+1:

Misiu commented:
Any chances to have this pulled?

vampolo commented:
Can this pull request be merged please ?

croeck commented:
+1

KoltonG commented:
Someone needs to get this merged! Its awesome!!

dsmorse commented:
Looks great I merged it into my fork dsmorse/gridster.js

rlagkrdy commented:
It's awesome!!, Thank U very much!!!!!! +100!!!
